src/containers/Cfe/index.tsx

```diff
@@ -31,20 +34,65 @@ interface CfeProps {
     viewerHeader: { cellId: string; label: string; value: string };
 }
 
-class Cfe extends React.Component<CfeProps> {
+interface CfeState {
+    defaultActiveKey: string[];
+    dontShowSmallScreenWarningAgain: boolean;
+    openKeys: string[];
+    width: number;
+    currentTab: string;
+}
+
+class Cfe extends React.Component<CfeProps, CfeState> {
     private static panelKeys = ["groupings", "selections"];
-    public state = {
+    public state: CfeState = {
         defaultActiveKey: [Cfe.panelKeys[0]],
         dontShowSmallScreenWarningAgain: false,
         openKeys: [Cfe.panelKeys[0]],
         width: window.innerWidth,
+        currentTab: PLOT_TAB_KEY,
     };
 
     public componentDidMount = () => {
         this.props.setShowSmallScreenWarning(window.innerWidth <= SMALL_SCREEN_WARNING_BREAKPOINT);
         window.addEventListener("resize", this.updateDimensions);
     };
 
+    public componentDidUpdate = (prevProps: CfeProps, prevState: CfeState) => {
+        const { currentTab } = this.state;
+        // Add selected cell info on top left corner of canvas
+        // TODO: The info should ideally be passed into web-3d-cell-viewer and
+        // handled there.
+        const { viewerHeader } = this.props;
+        const { viewerHeader: prevViewerHeader } = prevProps;
+        if (viewerHeader.cellId !== undefined && viewerHeader.cellId !== prevViewerHeader.cellId) {
+            const parent = document.querySelector(".cell-viewer-wrapper .ant-layout-content");
+            let wrapper = document.querySelector(".image-info-wrapper");
+            if (parent) {
+                if (!wrapper) {
+                    wrapper = document.createElement("div");
+                    wrapper.className = "image-info-wrapper";
+                }
+                parent.appendChild(wrapper);
+                // FIXME: the CSS Modules class names here aren't working, probably
+                // need to turn them into regular class names and use global selectors
+                // in the stylesheet.
+                wrapper.innerHTML = `
```

Comment:
This is obviously hackish.  Is there a way to do this as an overlay that stays within the react framework? or is the problem that it depends on knowing too much about internals of the viewer.  
I recommend making the stuff inside the if statement on line 67 (lines 68-87) into a function called something like "hackOverlayOn3DViewer".